langroid/agent/chat_agent.py

```diff
@@ -581,6 +655,108 @@ def enable_message(
             self.system_tool_format_instructions = self.tool_format_rules()
         self.system_tool_instructions = self.tool_instructions()
 
+    def _requests_and_tool_settings(self) -> tuple[Optional[set[str]], bool, bool]:
+        """
+        Returns the current set of enabled requests for inference and tools configs.
+        Used for restoring setings overriden by `set_output_format`.
+        """
+        return (
+            self.enabled_requests_for_inference,
+            self.config.use_functions_api,
```

Comment:
should we include `self.config.use_tools_api`?

Reply:
No need, as this isn't overridden (since we always disable `use_functions_api` which disables `use_tools_api` when we enable an output format).